### ISSUE TITLE
The value above the chart cannot be displayed

### ISSUE THREAD

**Issue by VcotyQin** [PhilJay/MPAndroidChart#3947]
setDrawValues() is not working.
When there is one item in the chart that is 0,the chart can work.but does not display when there is no zero data in the chart.
[HorizontalBarChart]

**Comment by almic** [comment]
Could you rephrase: "but that does not display when there is no zero data in the chart?"
It confuses me 😖 

If you have no data in the chart, then the chart skips drawing any data inside. But if you have a BarChart, and try to draw 0, you shouldn't see anything anyway. This makes sense to me. Instead of trying to draw a single bar with a value of `0`, I suggest not putting that data inside the chart at all, and instead use...

`chart.setNoDataText()`

...to inform the user that there was nothing to populate the chart with.

**Comment by VcotyQin** [comment]
Sorry,I'll tell it again.
If there is 0 in a set of data, it will show up like this : [https://raw.githubusercontent.com/VcotyQin/VaingloryGoData/master/another/2.jpeg](url)
If there is no 0 in a set of data,it will show up like this : [https://raw.githubusercontent.com/VcotyQin/VaingloryGoData/master/another/1.jpeg](url)

**Comment by almic** [comment]
I don't see why that would cause values to not be drawn at all. Could you please double-check your value formatters and chart initializers? If you still can't figure it out, then post the code here where you create the chart and add the data.

**Comment by almic** [comment]
@VcotyQin, did you figure this out?

**Comment by VcotyQin** [comment]
no

**Comment by almic** [comment]
I have a feeling this is because the library automatically hides labels in circumstances where they would start to overlap or be difficult to tell apart. This may be one of those cases. I'm still going through the full library code, if I see any red flags that could cause this I'll be sure to fix it.
